src/bazel/bazel_utils.ts

```diff
@@ -94,17 +94,21 @@ function shouldIgnorePath(fsPath: string): boolean {
 }
 
 /**
- * Search for the path to the directory that has the Bazel WORKSPACE file for
- * the given file.
+ * Search for the path to the directory that has the specified target file,
+ * starting at fsPath.
  *
- * If multiple directories along the path to the file have files called
- * "WORKSPACE", the lowest path is returned.
+ * If multiple directories along the path to the file have matching files,
+ * the lowest path is returned.
  *
- * @param fsPath The path to a file in a Bazel workspace.
- * @returns The path to the directory with the Bazel WORKSPACE file if found,
- * others undefined.
+ * @param fsPath The path to a file to use as a starting point for the search.
+ * @param targetFiles The names of the files to look for
+ * @returns The path to the directory where the target file is found, or
+ * undefined if the target file is not found.
  */
-export function getBazelWorkspaceFolder(fsPath: string): string | undefined {
+function getTargetFolderForFile(
```

Comment:
The word `target` is already overloaded in the Bazel ecosystem. How about `findAncestorDirectoryContaining` or something similar that better reflects the intention?